server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/V42417HistoricalPartitionPlanMigrate.java

```diff
@@ -0,0 +1,318 @@
+        JdbcOperationsUtil.batchCreate(jdbcTemplate, entities, sql, valueGetterMap, PartitionPlanTableEntity::setId);
+        log.info("[2/4] Partition plan table migration completed, "
+                + "tableName=partitionplan_table, entityCount={}", entities.size());
+    }
+
+    private void batchCreatePartiPlanTableKeys(JdbcTemplate jdbcTemplate,
+            List<PartitionPlanTablePartitionKeyEntity> entities) {
+        String sql = InsertSqlTemplateBuilder.from("partitionplan_table_partitionkey")
+                .field(PartitionPlanTablePartitionKeyEntity_.partitionKey)
+                .field(PartitionPlanTablePartitionKeyEntity_.strategy)
+                .field(PartitionPlanTablePartitionKeyEntity_.partitionplanTableId)
+                .field(PartitionPlanTablePartitionKeyEntity_.partitionKeyInvoker)
+                .field(PartitionPlanTablePartitionKeyEntity_.partitionKeyInvokerParameters)
+                .build();
+        ValueGetterBuilder<PartitionPlanTablePartitionKeyEntity> valueGetter = new ValueGetterBuilder<>();
+        List<Function<PartitionPlanTablePartitionKeyEntity, Object>> getter = valueGetter
+                .add(PartitionPlanTablePartitionKeyEntity::getPartitionKey)
+                .add(p -> p.getStrategy().name())
+                .add(PartitionPlanTablePartitionKeyEntity::getPartitionplanTableId)
+                .add(PartitionPlanTablePartitionKeyEntity::getPartitionKeyInvoker)
+                .add(PartitionPlanTablePartitionKeyEntity::getPartitionKeyInvokerParameters)
+                .build();
+        Map<Integer, Function<PartitionPlanTablePartitionKeyEntity, Object>> valueGetterMap = new HashMap<>();
+        IntStream.range(1, getter.size() + 1).forEach(i -> valueGetterMap.put(i, getter.get(i - 1)));
+        JdbcOperationsUtil.batchCreate(jdbcTemplate, entities, sql, valueGetterMap,
+                PartitionPlanTablePartitionKeyEntity::setId);
+        log.info("[3/4] Partition plan table migration completed, "
+                + "tableName=partitionplan_table_partitionkey, entityCount={}", entities.size());
+    }
+
+    private void batchUpdateScheduleParameterJson(JdbcTemplate jdbcTemplate,
+            List<PartitionPlanEntityWrapper> wrappers) {
+        List<Object[]> params = wrappers.stream().map(wrapper -> {
+            PartitionPlanConfig config = mapToModel(wrapper);
+            return new Object[] {JsonUtils.toJson(config), wrapper.getScheduleId()};
+        }).collect(Collectors.toList());
+        jdbcTemplate.batchUpdate("update schedule_schedule set job_parameters_json=? where id=?", params);
+        log.info("[4/4] Zoning plan parameter revision completed, tableName=schedule_schedule");
+    }
+
+    private PartitionPlanConfig mapToModel(PartitionPlanEntityWrapper wrapper) {
+        PartitionPlanConfig parameter = new PartitionPlanConfig();
+        parameter.setErrorStrategy(TaskErrorStrategy.ABORT);
```

Comment:
Why do we set these default values here? Where are these values from?

Reply:
this is for historical partition plan. this is our creation logic before :
![截屏2024-03-18 14 53 14](https://github.com/oceanbase/odc/assets/22992296/eea4acae-b5d8-4d30-b863-68f18f42290b)

as you can see, we will create such a database change task before, so that I migrate them and keep the same with the before~